### PR TITLE
pb-2009: Add /tmp as default director for the kopia cache and log dir.

### PR DESCRIPTION
**Pull request by siva-portworx**
**What this PR does / why we need it**:
```
    pb-2009: Add /tmp as default director for the kopia cache and log dir.

        - Also added /tmp/kopiaconfig for default config that will be
          used. These changes are done as in some platform, we do not
          have permission to create dir or file under root dir.
        - In the case of retry returning actual error instead of
          task.DoRetryWithTimeout function error.
```
**Which issue(s) this PR fixes** (optional)
Closes # pb-2009

**Special notes for your reviewer**:
Tested it on the OCS setup.

### CODE REVIEW

File: pkg/kopia/command.go [hunk 1]
```diff
@@ -88,6 +91,12 @@ func (c *Command) CreateCmd() *exec.Cmd {
 			c.Password,
 			"--prefix",
 			c.RepositoryName,
+			"--log-dir",
+			"logDir",
```
Comment (diptiranjanpx) [inline pkg/kopia/command.go:95]: these are variables , right? not fixed strings.

Comment (siva-portworx) [reply pkg/kopia/command.go:95]: Oops correct. I missed it. Thanks.

**Review by diptiranjanpx — APPROVED** [review]

File: pkg/kopia/command.go (hunk 1 above)
Comment (siva-portworx) [reply pkg/kopia/command.go]: fixed

File: pkg/kopia/command.go [hunk 2]
```diff
@@ -88,6 +91,12 @@ func (c *Command) CreateCmd() *exec.Cmd {
 			c.Password,
 			"--prefix",
 			c.RepositoryName,
+			"--log-dir",
+			logDir,
+			"--cache-directory",
```
Comment (prashanthpx) [inline pkg/kopia/command.go:96]: As this is common, could have been added here argsSlice = append(argsSlice, c.Flags...).

File: pkg/kopia/command.go [hunk 3]
```diff
@@ -140,6 +161,10 @@ func (c *Command) BackupCmd() *exec.Cmd {
 	argsSlice := []string{
 		"snapshot",
 		c.Name, // create command
+		"--log-dir",
+		logDir,
```
Comment (prashanthpx) [inline pkg/kopia/command.go:165]: It's not needed for backupcmd and as we are already passing it when the repository is created.
Same for restore command.
In general, we need to add for the following only
1. Repo create
2. Repo connect
Every other command uses repo create so it should be taken care of.

Comment (siva-portworx) [reply pkg/kopia/command.go:165]: As discussed, while testing the log-dir permission issue seen even when the repo connect went through the log-dir option. So added it for all the command. Only the cache-dir option is valid for the repo create and repo connect.

File: pkg/kopia/command.go (hunk 2 above)
Comment (siva-portworx) [reply pkg/kopia/command.go:96]: Followed like other options (config and password). Will take up the cleanup in next PR.

**Pull request merged**